src/Tree/HydroTree.cpp

```diff
@@ -1449,17 +1449,63 @@ void HydroTree<ndim,ParticleType>::FindMpiTransferParticles
     Box<ndim>& nodebox = mpinodes[inode].domain;
 
     // Start from root-cell
-    int NumPartFound = tree->FindBoxOverlapParticles(nodebox, all_particles_to_export,
+    tree->FindBoxOverlapParticles(nodebox, particles_to_export[inode],
                                                      hydro->GetParticleArray()) ;
 
     // Copy particles to per processor array
-    if (NumPartFound > 0)
-      particles_to_export[inode].insert(particles_to_export[inode].end(),
-                                        all_particles_to_export.end() - NumPartFound,
-                                        all_particles_to_export.end());
+    all_particles_to_export.insert(all_particles_to_export.end(),
+                                        particles_to_export[inode].begin(),
+                                        particles_to_export[inode].end());
   }
   //-----------------------------------------------------------------------------------------------
 
+#ifndef NDEBUG
+  VerifyUniqueIds(all_particles_to_export.size(),hydro->Nhydro,&all_particles_to_export[0]);
```

Comment:
This validation code is quite expensive. It is currently guarded by NDEBUG, so it will not be compiled with DEBUG_LEVEL=0. We have DEBUG_LEVEL=2 (macro VERIFY_ALL), but at the moment the code doesn't even compile with it. Besides, unless we are going to use it on a regular basis (e.g. on Travis), it's useless. What shall we do? Shall I open a separate issue for this?